torch/csrc/jit/passes/peephole.cpp

```diff
@@ -188,16 +188,24 @@ void PeepholeOptimizeImpl(Block* block, bool addmm_fusion_enabled) {
       // replace the output with true if node is __isnot__ or false if node is
       // __is__
       AT_ASSERT(node->inputs().size() == 2);
-      for (size_t check_none_index : {0, 1}) {
-        bool input_must_be_none =
-            node->inputs().at(check_none_index)->mustBeNone();
-        bool other_must_not_be_none =
-            node->inputs().at(1 - check_none_index)->mustNotBeNone();
-        if (input_must_be_none && other_must_not_be_none) {
-          WithInsertPoint guard(node);
-          auto output = node->owningGraph()->insertConstant(
-              node->kind() == aten::__isnot__);
-          node->output()->replaceAllUsesWith(output);
+      if (node->inputs().at(0)->mustBeNone() &&
+          node->inputs().at(1)->mustBeNone()) {
+        WithInsertPoint guard(node);
+        auto output =
+            node->owningGraph()->insertConstant(node->kind() == aten::__is__);
+        node->output()->replaceAllUsesWith(output);
+      } else {
+        for (size_t check_none_index : {0, 1}) {
```

Comment:
I can't follow this code after a cursory reading. This needs simplification or comments.

Reply:
It's unneeded now.